doc/source/tutorial/stats/outliers.md

```diff
@@ -0,0 +1,521 @@
+
+```{code-cell} ipython3
+mask = (x >= low) & (x <= high)
+x_trimmed = x[mask]
+x_trimmed
+```
+
+Whether threshold values are kept or trimmed depends on the inequality sign.
+
+```{code-cell} ipython3
+mask = (x > low) & (x < high)
+x[mask]
+```
+
+In 2-D, this approach *may* be possible if the number of elements retained in each row happen to be identical. Note that boolean indexing results in a 1-D array:
+
+```{code-cell} ipython3
+mask = (x2d >= low) & (x2d <= high)
+x2d[mask]
+```
+
+*When the number of elements retained in each row are identical (and only then)*, it is safe to reshape the output to an array with the original number of rows.
+
+```{code-cell} ipython3
+x2d[mask].reshape(x2d.shape[0], -1)  # -1 means "infer from the number of elements"
+```
+
+The more general option is to keep the trimmed elements in their original locations but to *mask* them. Note that the mask of elements to *remove* is the logical NOT of the elements to be kept.
+
+```{code-cell} ipython3
+mask2d = (x2d < low2d) | (x2d > high2d)
+x2d_masked = mxp.asarray(x2d, mask=mask2d)
+print(x2d_masked)
+```
+
+Another option is to replace the data to be trimmed with NaNs, which sometimes act as a "sentinel value" to denote that the data is missing.
+
+```{code-cell} ipython3
+x2d_nan = x2d.copy()
+x2d_nan[mask2d] = np.nan
+x2d_nan
+```
+
+Note the downsides to this approach:
+
+- This is only possible for data with floating point types, since integral types do not have a representation of NaN.
+- NaNs are primarily used to represent the result of an invalid operation such as `0/0`, which may lead to ambiguities.
+- There may be a performance impact, depending on the statistic.
+
++++
+
+# Computing the statistic
+
++++
+
+When the data is winsorized or trimmed data has actually been removed, working with it is no different than any other data.
+
+```{code-cell} ipython3
+np.mean(x_trimmed)
+```
+
+```{code-cell} ipython3
+np.mean(x_winsorized)  # result happens to be identical *for this data*
+```
+
+```{code-cell} ipython3
+np.mean(x2d_winsorized, axis=-1)  # results happen to be identical *for this data*
+```
+
+When the trimmed values are represented as masked elements of a masked array, only slightly more care is needed. For simple statistics like `mean` and `var`, use the corresponding function of the masked array namespace, `mxp`, rather than the NumPy namespace.
+
+```{code-cell} ipython3
+mxp.mean(x2d_masked, axis=-1)
+```
+
+Many SciPy functions also natively support masked arrays (see the documentation of the desired function).
+
+```{code-cell} ipython3
+stats.kurtosis(x2d_masked, axis=-1)
+```
+
+When the trimmed values are represented with NaNs, there are some NumPy functions that ignore NaNs:
+
+```{code-cell} ipython3
+np.nanmean(x2d_nan, axis=-1)
+```
+
+And many SciPy functions that accept a `nan_policy` argument.
+
+```{code-cell} ipython3
+stats.kurtosis(x2d_nan, nan_policy='omit', axis=-1)
+```
+
+# Recipes
```

Comment:
Things sort of got sloppy here. I only realized what a jungle it was after starting to write. Not sure whether I should try to make this more comprehensive, remove it entirely, or something in-between. Ideally, I would be able to identify a correspondence between options like `proportiontocut`, `relative`, and `inclusive` and the new method, but I find the existing behavior somewhat confusing, so I'm not sure if it's useful to map them to the new options.